src/Microsoft.IdentityModel.Clients.ActiveDirectory/Internal/Flows/AcquireTokenHandlerBase.cs

```diff
@@ -130,45 +138,38 @@ protected AcquireTokenHandlerBase(RequestData requestData)
                     ResultEx = this.tokenCache.LoadFromCache(CacheQueryData, this.CallState);
                     extendedLifetimeResultEx = ResultEx;
 
-                    if (ResultEx != null && ResultEx.Result != null)
+                    if (ResultEx?.Result != null &&
+                        ((ResultEx.Result.AccessToken == null && ResultEx.RefreshToken != null) ||
+                         (ResultEx.Result.ExtendedLifeTimeToken && ResultEx.RefreshToken != null)))
                     {
-                        if ((ResultEx.Result.AccessToken == null && ResultEx.RefreshToken != null) ||
-                            (ResultEx.Result.ExtendedLifeTimeToken && ResultEx.RefreshToken != null))
+                        ResultEx = await this.RefreshAccessTokenAsync(ResultEx).ConfigureAwait(false);
+                        if (ResultEx != null && ResultEx.Exception == null)
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this code to not nest more than 3 control flow statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://iddp-sonarqube:9000/coding_rules#rule_key=csharpsquid%3AS134)